base/test/archivespacemanager_tests.cpp

```diff
@@ -0,0 +1,170 @@
+#include "stdafx.h"
+#include <boost/test/unit_test.hpp>
+#include <boost/filesystem.hpp>
+#include "ExternalSourceModule.h"
+#include "ExternalSinkModule.h"
+#include "ArchiveSpaceManager.h"
+#include "Logger.h"
+#include "AIPExceptions.h"
+#include "test_utils.h"
+#include "Module.h"
+#include "PipeLine.h"
+
+BOOST_AUTO_TEST_SUITE(archivespacemanager_tests)
+
+void createDiskFiles ()
+{
+	boost::filesystem::create_directories(boost::filesystem::path("data/archive/Cam1/2022/10/01"));
+	boost::filesystem::create_directories(boost::filesystem::path("data/archive/Cam2/2022/10/01"));
+	boost::filesystem::create_directories(boost::filesystem::path("data/archive/Cam3/2022/10/01"));
+	boost::filesystem::ofstream a(boost::filesystem::path("data/archive/Cam1/2022/10/01/file0.txt"));
+	for (int i = 0; i < 10023; i++)
+	{
+		a << "a";
+	}
+	a.close();
+	boost::filesystem::ofstream b(boost::filesystem::path("data/archive/Cam2/2022/10/01/file0.txt"));
+	for (int i = 0; i < 10023; i++)
+	{
+		b << "b";
+	}
+	b.close();
+	boost::filesystem::ofstream c(boost::filesystem::path("data/archive/Cam3/2022/10/01/file0.txt"));
+	for (int i = 0; i < 10023; i++)
+	{
+		c << "c";
+	}
+	c.close();
+};
+
+BOOST_AUTO_TEST_CASE(basic)
+{
+	createDiskFiles();
+	std::vector<std::string> folder = { "./data/archive" };
+	Test_Utils::FileCleaner f(folder);
+	uint32_t lowerWaterMark = 25000;
+	uint32_t upperWaterMark = 30000;
+	auto source = boost::shared_ptr<ExternalSourceModule>(new ExternalSourceModule());
```

Comment:
we dont need an external source module to decide whien to trigger it... it runs on it's own.

Reply:
External Source module is removed and ArchiveSpaceManager is made source.